src/migrations/20240420000000-correct-onApprovedAR-values.js

```diff
@@ -0,0 +1,331 @@
+const {
+  prepMigration,
+} = require('../lib/migration');
+
+module.exports = {
+  up: async (queryInterface, Sequelize) => queryInterface.sequelize.transaction(
+    async (transaction) => {
+      await prepMigration(queryInterface, transaction, __filename);
+      await queryInterface.sequelize.query(/* sql */`
+
+      --  1. Calculate correct onApprovedAR values for objectives
+      DROP TABLE IF EXISTS objectives_on_ars;
+      CREATE TEMP TABLE objectives_on_ars
+      AS
+      SELECT
+        o.id oid,
+        BOOL_OR(ar.id IS NOT NULL AND ar."calculatedStatus" = 'approved') on_approved_ar,
+        BOOL_OR(ar.id IS NOT NULL) on_ar
+      FROM "Objectives" o
+      LEFT JOIN "ActivityReportObjectives" aro
+        ON o.id = aro."objectiveId"
+      LEFT JOIN "ActivityReports" ar
+        ON aro."activityReportId" = ar.id
+        AND ar."calculatedStatus" != 'deleted'
+      GROUP BY 1
+      ;
+      --  2. Calculate correct onApprovedAR values for goals
+      DROP TABLE IF EXISTS goals_on_ars;
+      CREATE TEMP TABLE goals_on_ars
+      AS
+      SELECT
+        g.id gid,
+        BOOL_OR(
+          (ar.id IS NOT NULL AND ar."calculatedStatus" = 'approved')
+          OR
+          COALESCE(ooaa.on_approved_ar,FALSE)
```

Comment:
I haven't gotten to the usage of this table yet. Is there any concern that a goal without an ar would have on_approved_ar set to true with the ooaa table link?

Reply:
If the Objective is on an AR and is linked to the Goal, then it is inherently on an AR. Basically, there's two ways to link a goal to an AR: directly through activityReportGoals->Goals, or indirectly through activityReportObjectives->Objectives->Goals